### PR TITLE
Document filling of Poly3DCollection

### LINKED ISSUE #16633: Wrong drawing Poly3DCollection
Drawing 3D polygon with the code:

```
fig = plt.figure()
ax = fig.add_subplot(projection='3d')
ax.set_xlim((0, 5))
ax.set_ylim((0, 5))
ax.set_zlim((0, 5))
x = [1,2,3, 3,2,1]
y = [2,1,2, 2,1,2]
z = [1,1,1, 2,2,2]
v = [list(zip(x, y, z))]
pc = Poly3DCollection(v)
ax.add_collection3d(pc)
```

Leads to the following result on scene rotation:

While the polygon should looks like this:

**Matplotlib version**
  * Operating system: Ubuntu 16.04.6 LTS
  * Matplotlib version: 3.1.2
  * Matplotlib backend (`print(matplotlib.get_backend())`): module://ipympl.backend_nbagg
  * Python version: 3.8.1
  * Jupyter version (if applicable): 1.0.0
  * Other libraries:

### PR DESCRIPTION
**Pull request by timhoffm**
## PR Summary

Closes #16633, by documenting the limitations and linking to `plot_trisurf` as a better way to create filled surfaces.

### CODE REVIEW

File: lib/mpl_toolkits/mplot3d/art3d.py [hunk 1]
```diff
@@ -530,17 +530,39 @@ def patch_collection_2d_to_3d(col, zs=0, zdir='z', depthshade=True):
...
+        Parameters
+        ----------
+        verts : list of array-like Nx3
+            Each element describes a polygon as a sequnce of ``N_i`` points
+            ``(x, y, z)``.
+        zsort : {'average', 'min', 'max'}, default: 'average'
+            The calculation method for the z-order.
+            See `~.Poly3DCollection.set_zsort` for details.
+        *args, **kwargs
+            All other parameters are forwarded to `.PolyCollection`.
```
Comment (eric-wieser) [inline lib/mpl_toolkits/mplot3d/art3d.py]: What does `.` mean at the beginning of a reference? Is this a sphinx thing, or something specific to matplotlib?

Comment (ImportanceOfBeingErnest) [reply lib/mpl_toolkits/mplot3d/art3d.py]: It's a "sphinx thing". But there is documentation on it in the matplotlib dev guide: https://matplotlib.org/devdocs/devel/documenting_mpl.html#referring-to-other-code

Comment (timhoffm) [reply lib/mpl_toolkits/mplot3d/art3d.py]: See also https://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html#cross-referencing-syntax

File: lib/mpl_toolkits/mplot3d/art3d.py [hunk 2]
```diff
@@ -530,17 +530,39 @@ def patch_collection_2d_to_3d(col, zs=0, zdir='z', depthshade=True):
...
+        unless the polygon is planar.
+
+        In practice, Matplotlib performs the filling on the 2D projection of
+        the polygon. This gives a correct filling appearance only for planar
+        polygons. For all other polygons, you'll find orientations in which
+        the edges of the polygon intersect in the projection. This will lead
+        to an incorrect visualization of the 3D area.
+
+        If you need filled areas, it is recommended to create them via
+        `~mpl_toolkits.mplot3d.axes3d.Axes3D.plot_trisurf`, which creates a
```
Comment (anntzer) [inline lib/mpl_toolkits/mplot3d/art3d.py]: I would just write `.Axes3D.plot_trisurf` (or `~.Axes3D.plot_trisurf`).
Can self-merge with or without that change.

**Pull request merged**